pkg/scheduler/plugins/gang/gang.go

```diff
@@ -162,9 +162,21 @@ func (gp *gangPlugin) OnSessionClose(ssn *framework.Session) {
 	var unScheduleJobCount int
 	for _, job := range ssn.Jobs {
 		if !job.Ready() {
-			unreadyTaskCount = job.MinAvailable - job.ReadyTaskNum()
+			schedulableTaskNum := func() (num int32) {
+				for _, task := range job.TaskStatusIndex[api.Pending] {
+					ctx := task.GetTransactionContext()
+					if task.LastTransaction != nil {
+						ctx = *task.LastTransaction
+					}
+					if api.AllocatedStatus(ctx.Status) {
+						num++
+					}
+				}
+				return num + job.ReadyTaskNum()
```

Comment:
why `+job.ReadyTaskNum()` here, can you elaborate more ?

Reply:
- `ReadyTask`s are tasks already scheduled, maybe already bound. They alone with pending but allocatable tasks are all schedulable tasks for `gang`.

Reply:
Is it possible to have some repeated calculation since it actually contains the task states at two moments

Reply:
It won't. `Pending` tasks can never be tasks of `AllocatedStatus`.

Reply:
Think of one situation:
minAvailable: 4, total tasks 5, the first four tasks are allocated, then the entire job is committed(first four tasks become running), and the remaining one task is pending, then the calculation here will repeat the calculation of the first four tasks, right?

Reply:
It won't. The first four tasks are running, they are calculated in `ReadyTaskNum()`. The remaining pending task will be calculated in `job.TaskStatusIndex[api.Pending]` in the above codes, hence, as i said, it won't be added twice.

Reply:
ahh got you, missed job.TaskStatusIndex[**api.Pending**].. : Dazzled today)